

Discord Bot Url:

https://discord.com/api/oauth2/authorize?client_id=1090369592993533993&permissions=154619341888&scope=bot

Video timestamp 2/28/2023: 4:36

ToDo for 4.9.23: 
        Check to see if encouragements db on replit is update or not, when user tries to add new message.
        If not figure out why.
ToDO for 4.12.23:
        Try to figure out why delete clause in on message is: 
        a. not deleting records(is it not parsing the index given by the user?(line 86)  

        

        , is the delete_encrouagement function faulty (line 87), something else entirely) 
        and 
        b. returning an empty array.  

-- follow up 12:02PM 4.13.23: 
        1. the statement on line 86 which parses the message given by user to a valid index, works when run in isolation

        2. When I stepped through the code (On Replit) with the debugger the if statement on line 85 
           "if 'encouragments' in db.keys():" seems to resolve to False, bybassing the block of code in said if statement
           lines 86-88 aren't running. So that's a good take away.


ToDo for 4.14.23:
        Try to figure out the meaning of the if statement on line 85 "if 'encouragments' in db.keys():", 
        and why it evaluates to false. 

        Try to get the code block (lines 85-88) to run.
-- follow up 3:37PM 4.14.23:
        1. When when I checked for which keys were in the replit data base, using the python shell command used was (print()db.keys), it returned
           {'encouragements'}. When I checked to see the values in the database using (print(db.get('encouragements'))) it gave me my list of user 
           added encouragements. So this throws my earlier hypothosis in to question. When I enter the statement ('encouragments' in db.keys()) 
           into the same python shell it still returns false.

           UPDATE 3:55PM 4.14.23: 'encouragements' was spelled incorrectly in a few different places inside the 'if msg.startswith("$del"):' 
           clause of the on_message function! So I changed 'encouragments' to 'encouragements', and now it works!

ToDo for 4.17-18.23:
        After indenting if statement on line 78 and replacing the argument starter_encouragements in random.choice() with options;
        when responding is set to true, instead of responding to a message contianing a sad word, with an encouraging message 
        discord bot returns this error in the replit console:

                Traceback (most recent call last):
                File "/home/runner/Discord-bot/venv/lib/python3.10/site-packages/discord/client.py", 
                line 441, in _run_event
                await coro(*args, **kwargs)
                File "main.py", line 79, in on_message
                await message.channel.send(random.choice(options))
                File "/nix/store/hd4cc9rh83j291r5539hkf6qd8lgiikb-python3-3.10.8/lib/python3.10/random.py",
                line 378, in choice
                return seq[self._randbelow(len(seq))]
                TypeError: object of type 'NoneType' has no len()

        Figure out why.